src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1428,6 +1466,17 @@ impl Decimal {
         Res::Ok(x)
     }
 
+    /// `as_i64_with_ctx` returns int part of the decimal.
+    pub fn as_i64_with_ctx(&self, ctx: &EvalContext) -> ::std::result::Result<i64, ExprError> {
+        let res = self.as_i64();
+        if ctx.ignore_truncate || ctx.truncate_as_warning {
+            if let Res::Truncated(ref i) = res {
```

Comment:
Could we use `convert::handle_truncate` here since one day we may need to process truncate as warning?